sites/svelte-5-preview/src/routes/docs/content/01-api/02-runes.md

```diff
@@ -159,48 +159,45 @@ In essence, `$derived(expression)` is equivalent to `$derived.by(() => expressio
 
 ```svelte
 <script>
-	let count = $state(0);
-	let doubled = $derived(count * 2);
+	let size = $state(50);
+	let color = $state('#ff3e00');
+
+	let canvas;
 
 	$effect(() => {
-		console.log({ count, doubled });
+		const context = canvas.getContext('2d');
+		context.clearRect(0, 0, canvas.width, canvas.height);
+
+		// this will re-run whenever `color` or `size` change
+		context.fillStyle = color;
+		context.fillRect(0, 0, size, size);
 	});
 </script>
 
-<button on:click={() => count++}>
-	{doubled}
-</button>
-
-<p>{count} doubled is {doubled}</p>
+<canvas bind:this={canvas} width="100" height="100" />
 ```
 
-`$effect` will automatically subscribe to any `$state` or `$derived` values it reads _synchronously_ and reruns whenever their values change — that means, values after an `await` or inside a `setTimeout` will _not_ be tracked. `$effect` will run after the DOM has been updated.
+The function passed to `$effect` will run when the component mounts, and again — after any updates to the DOM — whenever there are changes to any values it reads that were declared with `$state` or `$derived` (including those passed in with `$props`).
 
-```svelte
-<script>
-	let count = $state(0);
-	let doubled = $derived(count * 2);
-
-	$effect(() => {
-		// runs after the DOM has been updated
-		// when the component is mounted
-		// and whenever `count` changes,
-		// but not when `doubled` changes,
-		console.log(count);
-
-		setTimeout(() => console.log(doubled));
-	});
-</script>
+Values that are read asynchronously — after an `await` or inside a `setTimeout`, for example — will _not_ be tracked. Here, the canvas will be repainted when `color` changes, but not when `size` changes ([demo](/#H4sIAAAAAAAAE31T24rbMBD9lUG7kCxsbG_LvrhOoPQP2r7VhSjy2BbIspHGuTT436tLnMtSCiaOzpw5M2dGPrNaKrQs_3VmmnfIcvZ1GNgro9PgD3aPitCdbT8a4ZHCCiMH2pS6JIUEVv5BWMOzJU64fM9evswR0ave3EKLp7r-jFm2iIwri-s9tx5ywDPWNQpaLl9gvYFz4JHotfVqmvBITi9mJA3St4gtF5-qWZUuvEQo5Oa7F8tewT2XrIOsqL2eWpRNS7eGSkpToFZaOEilwODKjBoOLWrco4FtsLQF0XLdoE2S5LGmm6X6QSflBxKod8IW6afssB8_uAslndJuJNA9hWKw9VO91pmJ92XunHlu_J1nMDk8_p_8q0hvO9NFtA47qavcW12fIzJBmM26ZG9ZVjKIs7ke05hdyT0Ixa11Ad-P6ZUtWbgNheI7VJvYQiH14Bz5a-SYxvtwIqHonqsR12ff8ORkQ-chP70T-L9eGO4HvYAFwRh9UCxS13h0YP2CgmoyG5h3setNhWZF_ZDD23AE2ytZwZMQ4jLYgVeV1I2LYgfZBey4aaR-xCppB8VPOdQKjxes4UMgxcVcvwHf4dzAv9K4ko1eScLO5iDQXQFzL5gl7zdJt-nZnXYfbddXspZYsZzMiNPv6S8Bl41G7wMAAA==)):
 
-<button on:click={() => count++}>
-	{doubled}
-</button>
-
-<p>{count} doubled is {doubled}</p>
+```diff
+$effect(() => {
+	const context = canvas.getContext('2d');
+	context.clearRect(0, 0, canvas.width, canvas.height);
+
+	// this will re-run whenever `color` changes...
+	context.fillStyle = color;
+
++	setTimeout(() => {
```

Comment:
Why is this a diff block? The comments are almost unreadable like that - I think it would be better to just make this a `svelte` code block.